tests/postgres_tests/test_indexes.py

```diff
@@ -234,6 +241,25 @@ def test_deconstruction(self):
         )
 
 
+class CustomIndexTests(IndexTestMixin, PostgreSQLSimpleTestCase):
+    index_class = DummyIndex
+
+    def test_suffix(self):
+        self.assertEqual(self.index_class.suffix, "dummy")
+
+    def test_deconstruction(self):
+        field_name = "dummy_field"
+        index_name = "dummy_index"
+        index = self.index_class(fields=[field_name], name=index_name)
+        path, args, kwargs = index.deconstruct()
+        self.assertEqual(
+            path,
+            "django.contrib.postgres.indexes.%s" % self.index_class.__name__,
+        )
+        self.assertEqual(args, ())
+        self.assertEqual(kwargs, {"fields": [field_name], "name": index_name})
```

Comment:
I don't see much value in this tests and adding `test_dummy_index()` to the schema tests. I'd rather add a separate class and define custom index in it, e.g.:
```python
class PostgresIndexTests(PostgreSQLTestCase):
    def test_custom_suffix_index(self):
        class CustomSuffixIndex(PostgresIndex):
            suffix = "custom"

            def create_sql(self, model, schema_editor, using="gin", **kwargs):
                 return super().create_sql(model, schema_editor, using=using, **kwargs)

        index = CustomSuffixIndex(fields=["field"], name="custom_suffix_idx")
        with connection.schema_editor() as editor:
            self.assertIn(
                " USING gin ",
                str(index.create_sql(IntegerArrayModel, editor)),
            )   

```

Reply:
In `tests/postgres_tests/test_indexes.py` each index class has its own `XIndexTests` which constructs the object, checks the properties etc. Then the `SchemaTests` implements one or two methods for each class to create the index and check the behavior against the schema editor. I followed the same pattern, figuring `DummyIndex` should get the same treatment: it's an index as any, albeit with less customisations.

Reply:
Other indexes tested in `test_indexes` are builtin indexes that we need to test. `DummyIndex` was created as a regression test for using the `using` argument of `create_sql()`, we don't need to cover it by multiple tests (6 currently). It's enough to add a single test that checks what this patch is intended to fix.

Reply:
Ok, what do you think of a method on `SchemaTests` with just:
```python
def test_custom_suffix(self):
    class CustomSuffixIndex(GinIndex):
        suffix = "sfx"

        def create_sql(self, model, schema_editor, using="gin", **kwargs):
            return super().create_sql(model, schema_editor, using=using, **kwargs)

    index = CustomSuffixIndex(fields=["field"], name="custom_suffix_idx")
    self.assertEqual(index.suffix, "sfx")
    with connection.schema_editor() as editor:
        self.assertIn(
            " USING gin ",
            str(index.create_sql(CharFieldModel, editor)),
        )
```

Reply:
> Ok, what do you think of a method on `SchemaTests` with just:

Looks good, do we need to subclass `GinIndex` instead of `PostgresIndex`? We want to fix an issue in `PostgresIndex`, so using `GinIndex` can be a bit misleading.

Reply:
See https://github.com/django/django/pull/15600/commits/55c0a4a5d2a1e98b9c678001410471f0acb6b658